flask_sqlalchemy/__init__.py

```diff
@@ -577,7 +577,7 @@ def _should_set_tablename(bases, d):
             return False
 
         for name in dir(base):
-            attr = getattr(base, name)
+            attr = getattr(base, name, None)
```

Comment:
I came across this when Flask-SQLAlchemy was accessing a class property reliant on the name of the subclass in an abstract superclass as the superclass.

This fix catches AttributeErrors that may arise from Flask-SQLAlchemy's handling of abstract mixins, but would never be thrown in the program at runtime. We could change the way it handles abstract superclasses, but this fix is six characters and I can't think of any other problems which might arise from its current behavior.